File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

n = int(sys.stdin.readline())
queue = deque([])
for _ in range(n):
    commands = sys.stdin.readline().rstrip().split()
    if len(commands) == 2:
        queue.append(int(commands[1]))
    else:
        if commands[0] == 'pop':
            print(-1 if not queue else queue.popleft())
        elif commands[0] == 'size':
            print(len(queue))
        elif commands[0] == 'empty':
            print(1 if not queue else 0)
        elif commands[0] == 'front':
            print(-1 if not queue else queue[0])
        elif commands[0] == 'back':
            print(-1 if not queue else queue[-1])

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <string>
#include <queue>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int n;
	cin >> n;
	string cmd;
	int x;
	queue<int> q;
	while (n--) {
		cin >> cmd;
		if (cmd == "push") {
			cin >> x;
			q.push(x);
		}
		else if (cmd == "pop") {
			if (q.empty()) cout << -1 << '\n';
			else {
				cout << q.front() << '\n';
				q.pop();
			}
		}
		else if (cmd == "size") {
			cout << q.size() << '\n';
		}
		else if (cmd == "empty") {
			cout << q.empty() << '\n';
		}
		else if (cmd == "front") {
			if (q.empty()) cout << -1 << '\n';
			else cout << q.front() << '\n';
		}
		else if (cmd == "back") {
			if (q.empty()) cout << -1 << '\n';
			else cout << q.back() << '\n';
		}
	}
}